app/containers/MassEnergizeSuperAdmin/ME  Tools/icon dialog/IconDialog.js

```diff
@@ -0,0 +1,230 @@
+import React, { useState } from "react";
+import "./IconDialog.css";
+import debounce from "lodash.debounce";
+const ICONS = require("./icon_files.json");
+const RECENT_ICONS_KEY = "RECENT_ICONS_KEY";
+const MAX_RECENTS = 10; // number of icons that should be kept in the recent list
+function IconDialog({
+  perPage = 50,
+  placeholder,
+  onIconSelected,
+  defaultValue,
+  value,
+  maxRecents,
+  recentItemsStorageKey,
+}) {
+  const icons = ICONS.ICON_FILES;
+  const [iconSet, setIconSet] = useState(icons.slice(0, perPage));
+  const [searched, setSearched] = useState([]);
+  const [text, setText] = useState("");
+  const pages = Math.round(icons.length / perPage);
+  const [page, setPage] = useState(1);
+  const [show, setShow] = useState(false);
+  const [selected, setSelected] = useState(defaultValue || value || null);
+  const maximumNumberOfRecentItems = maxRecents || MAX_RECENTS;
+  const iconsStorageKey = recentItemsStorageKey || RECENT_ICONS_KEY;
+  const storedRecents = JSON.parse(localStorage.getItem(iconsStorageKey));
+  const [recents, setRecents] = useState(storedRecents || []);
+
+  const nextPage = () => {
+    if (page >= pages) return;
+    const newStartIndex = perPage * page;
+    const endIndex = newStartIndex + perPage;
+    const nextSet = icons.slice(newStartIndex, endIndex);
+    setIconSet(nextSet);
+    setPage(page + 1);
+  };
+
+  const prevPage = () => {
+    if (page <= 1) return;
+    const endIndex = perPage * (page - 1);
+    const newStartIndex = endIndex - perPage;
+    const nextSet = icons.slice(newStartIndex, endIndex);
+    setIconSet(nextSet);
+    setPage(page - 1);
+  };
+
+  const search = (text) => {
+    setText(text);
+    if (!text) return setSearched([]);
+    const found = icons.filter((name) => name.includes(text));
+    setSearched(found);
+  };
+
+  const addToRecents = (iconName) => {
+    const rem = recents.filter((ic) => ic !== iconName);
+    var content = [iconName, ...rem];
+    if (content.length > maximumNumberOfRecentItems)
+      content = content.slice(0, maximumNumberOfRecentItems);
+    setRecents(content);
+    localStorage.setItem(iconsStorageKey, JSON.stringify(content));
```

Comment:
This means that frequently used icons would persist across sessions right? 

I'm also assuming we want recent icons to be per admin, so this is good.

Reply:
No.  Ideally we want the most common icons chosen for our community sites to be at the top of the list.  Not per admin.

Reply:
We could do it manually, updating a json file or something that the front end can read in to list commonly used icons at the top.